### DISCUSSION TITLE
Find in file alias / function

### DISCUSSION THREAD

**Discussion by SIMULATAN** [ohmyzsh/ohmyzsh#11329] [Ideas]
I often see myself searching in files with the `find` command, `cat`, and `grep`, but the command is quite difficult to remember.

Thus, I think it would be great to have a `fif` command in an optional plugin that does this for you.

Any proposals for the command to use? I'd PR the change once there's a full fitting command.

**Comment by mcornella** [comment]
Can you specify exactly what the command would do? `find` is used to search for files, while `grep` is used to find a specific string in the file contents (`cat` is not needed here: `grep string file`).

How would you combine both? If you are talking about something like, *find hello in all files ending with .txt*, you could use wildcards directly:

```sh
grep hello *.txt
```

**Reply by SIMULATAN** [reply]
Oh; yeah that solves my issue. I didn't know that grep could do that with wildcards and recursion, thanks a lot! That solves my issue.
Maybe it would help users to make a function like seen below as a plugin?
```bash
function fif() {
  grep -Ir "$@" *
}
```
For the record, the one I wanted to use was `alias fif="find . -type f -print | xargs grep"`, which does essentially the same thing, but with the find command and stuff, and looking back at it, I wonder if that even works in the first place.
`fif` stands for `find in files` by the way.
